j5/backends/console/arduino/uno.py

```diff
@@ -0,0 +1,113 @@
+"""Console Backend for the Arduino Uno."""
+from typing import Mapping, Optional, Set, Type
+
+from j5.backends import Backend
+from j5.backends.console import Console, ConsoleEnvironment
+from j5.boards import Board
+from j5.boards.arduino import ArduinoUnoBoard
+from j5.components import GPIOPinInterface, GPIOPinMode, LEDInterface
+
+
+class PinData:
+    """Contains data about a pin."""
+
+    mode: GPIOPinMode
+    digital_state: bool
+
+    def __init__(self, *, mode: GPIOPinMode, digital_state: bool):
+        self.mode = mode
+        self.digital_state = digital_state
+
+
+class ArduinoUnoConsoleBackend(GPIOPinInterface, LEDInterface, Backend):
+    """Console Backend for the Arduino Uno."""
+
+    environment = ConsoleEnvironment
+    board = ArduinoUnoBoard
+
+    @classmethod
+    def discover(cls) -> Set[Board]:
+        """Discover boards that this backend can control."""
+        raise NotImplementedError("The Console Backend cannot discover boards.")
+
+    def __init__(self, serial: str, console_class: Type[Console] = Console) -> None:
+        self._serial = serial
+
+        self._pins: Mapping[int, PinData] = {
+            i: PinData(mode=GPIOPinMode.DIGITAL_OUTPUT, digital_state=False)
+            for i in range(2, 20)
+            # Digital 2 - 13
+            # Analogue 14 - 19
```

Comment:
By the looks of things all pins here are set to `GPIOPinMode.DIGITAL_OUTPUT` by default. Should this comment about analogue pin ranges be removed, or am I missing something?

Reply:
Analogue Pins (A0-A5) on the Arduino Uno also support Digital write and read, so this is correct.

This backend doesn't store the state for the Analogue state of the pins, it is given through stdin.